### ISSUE TITLE
update-users-groups.pl creates duplicate uids

### ISSUE THREAD

**Issue by domenkozar** [NixOS/nixpkgs#14148]
To reproduce:

- set `mutableUsers` to `false`
- rebuild system with two users that don't have uid specified
- rebuild system by specifying uid of the first user to match the generated uid of the second user

They will now share the uid.

Expected behavior: error should be raised

cc @edolstra

**Comment by cleverca22** [comment]
this also happens if you force the uid for 1 user, and dont set the uid at all on another, i need to force the uid on every user to avoid collisions

**Comment by domenkozar** [comment]
@cleverca22 your issue should be fixed in e561edc322d275c3687fec431935095cfc717147, I'm still verifying if my problem is covered :)

**Comment by domenkozar** [comment]
There might be some other edge cases, but now this issue yields:

```
warning: not applying UID change of user ‘user’ (1000 -> 1001)
```
